pkg/trace/agent/agent.go

```diff
@@ -110,13 +113,14 @@ func (a *Agent) Run() {
 }
 
 func (a *Agent) work() {
+	sublayerCalculator := stats.NewSublayersCalculator(1000)
 	for {
 		select {
 		case t, ok := <-a.In:
 			if !ok {
 				return
 			}
-			a.Process(t)
+			a.Process(t, sublayerCalculator)
```

Comment:
Might want a reusableBuffer struct to pass.
There are other use cases like reusing bytes buffer for string hashing for the samplers.
wdyt @gbbr?

Reply:
I'm not sure what you mean here. Which buffer are you referring to?

Reply:
byte buffers used to hashs strings with fnv

Reply:
https://github.com/DataDog/datadog-agent/blob/master/pkg/trace/sampler/signature.go#L75-L93